### DISCUSSION TITLE
Merge transformation not working using uql

### DISCUSSION THREAD

**Discussion by Usama420-git** [grafana/grafana-infinity-datasource#499] [General]
Merge transformation not working using uql
`parse-json
| jsonata "$map(value,function($v){{ 'inviteDateTime': $v.$substring(inviteDateTime, 0, 10), 'id': $v.id, 'duration': $v.duration,  'successfulCall': $v.successfulCall}})"`
when I am using Json and adding columns the merge transformation is working but when I am using UQL query merge not working when 1 of the query returns empty but this is not the same case with json 

I want to use uql because I want to convert my date with jsonata. but why the merge not working when I am using UQL Query

![query](https://user-images.githubusercontent.com/51694624/209547808-d69ed0e8-9b48-4992-9be2-99e4d71eede7.PNG)

**Comment by Usama420-git** [comment]
@yesoreyeram any workaround for this?

**Reply by yesoreyeram** [reply]
Sorry.. Haven't investigated. In holidays.

**Reply by Usama420-git** [reply]
Ok enjoy your holidays 👍

**Reply by Usama420-git** [reply]
can we use substring in uql I tried like this

`| extend "inviteDateTime"=todatetime(substring("inviteDateTime", 0, 10))`

don't want to use jsonata because when array is empty jsonata creates a column name result with nothing in it and I have a problem while merging. and when using only this 

`| extend "inviteDateTime"=todatetime("inviteDateTime")
| extend "inviteDateTime"=startofday("inviteDateTime")`

getting problem of rounding it by day
like I have 90 days but this is showing only 86 that's why substring is important

whenever you are free please tell me what can I do?

**Reply by yesoreyeram** [reply]
current version of UQL doesn't have `substring`. instead you can use `extract` method as shown below which extract first ten characters using regex.

```kusto
| extend "inviteDateTime"=extract('^.{0,10}',0,"inviteDateTime")
| extend "inviteDateTime"=todatetime("inviteDateTime")
```

or you can use `split` method

```kusto
| extend "inviteDateTime"=split("inviteDateTime",'T')
| extend "inviteDateTime"="inviteDateTime[0]"
| extend "inviteDateTime"=todatetime("inviteDateTime")
```

**Reply by Usama420-git** [reply]
extract and split both are working for me thank you so much
